### PR TITLE
Add docs for Teleport Connect

### PR DESCRIPTION
**Pull request by ravicious**
[Rendered version](https://github.com/gravitational/teleport/blob/ravicious/connect-docs/docs/pages/use-teleport/teleport-connect.mdx).

I'm not a native speaker so please be sure to point out any grammatical mistakes!

Closes gravitational/webapps.e#349.

### CODE REVIEW

**Comment by ravicious** [comment]
@gzdunek At some point we'll have to update the docs to account for the Windows version.

**Review by gzdunek — APPROVED** [review]

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+![resources tab in Teleport Connect](../../img/use-teleport/connect-cluster.png)
+
+## Installation & Upgrade
+
+Head over to the [Downloads](https://goteleport.com/download/) page to download the most recent
+version of Teleport Connect. As of now, Teleport Connect supports macOS only. Double-click the
+downloaded .dmg file and drag the Teleport Connect icon to the Applications folder.
+
+To upgrade Teleport Connect to a newer version, download the .dmg file with that version and follow
+the installation steps.
```
Comment (stevenGravy) [inline docs/pages/use-teleport/teleport-connect.mdx:18]: Best to just say drag to the Applications folder

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+- The connection list (the top left) showing recent connections, allowing you to seamlessly switch
+  between them.
+- The command bar (in the middle) which launches any command you input in a new terminal tab.
+- The cluster selector (to the left of the command bar) shows up only if there are leaf clusters
+  connected to the root cluster. It lets you browse leaf cluster resources. It also changes
+  which cluster the commands in the command bar are going to target.
+
+## Connecting to an SSH Server
+
+1. Open a tab with cluster resources by clicking on the plus symbol at the end of the tab bar. You
```
Comment (avatus) [inline docs/pages/use-teleport/teleport-connect.mdx:37]: should read at the `right` end I think. Even tho there isn't one on the left, they might search there first unknowingly. Or something like 'the right most button' Idk how to word. But its on the right!

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+   can also press `Cmd + T` to achieve the same result.
+2. Look for the SSH server you want to connect to and click the Connect button to the right.
+3. Select or enter the login you wish to use and press `Enter`.
```
Comment (avatus) [inline docs/pages/use-teleport/teleport-connect.mdx:40]: "enter the login" should maybe be "select or enter the `ssh` user you wish to login as

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+4. A new tab will open with a shell session on the chosen server.
+
+Alternatively, you can type `tsh ssh` into the command bar and the autocompletion will guide you to
+connect to a server.
```
Comment (avatus) [inline docs/pages/use-teleport/teleport-connect.mdx:44]: will guide you "through the server connection process" or "will guide you through connecting to a server" or "will help you connect to a server"

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+After you add your first cluster, you can keep adding more by opening the profile selector in the
+top right and clicking at the *+ Add another cluster* button.
+
+At the moment Teleport Connect supports only one user per cluster. To log in as a different user,
+log out of the cluster first.
+
+## Restarting and Reconnecting
+
+Teleport Connect remembers the tabs that you had open and will ask you on restart if you want to
```
Comment (avatus) [inline docs/pages/use-teleport/teleport-connect.mdx:81]: Not sure if this change is needed, but maybe reword it around "the previous session" or something around the word session since the modal that pops up when you open says "start new session" or reopen. lets use the "session" wording somewhere. "Connect session" maybe?

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+logging in again.
+
+To force the app to log you out of all clusters, close the app and remove the `~/Library/Application
+Support/Teleport Connect/tsh` folder. Removing the file under `~/Library/Application
+Support/Teleport Connect/app_state.json` will clear all remembered tabs and connections.
+
+### Submitting an Issue
+
+To submit an issue, click the Submit Feedback button at the bottom right (the speech bubble symbol)
+and follow the Submit a Bug link.
```
Comment (avatus) [inline docs/pages/use-teleport/teleport-connect.mdx:100]: lets highlight or quote "Submit a Bug"

**Review by avatus — APPROVED** [review]

**Comment by avatus** [comment]
I'm not the best copy writer so if I suggested doesn't make sense, please push back

File: docs/pages/use-teleport/teleport-connect.mdx (hunk 5 above)
Comment (ravicious) [reply docs/pages/use-teleport/teleport-connect.mdx]: Yeah that makes sense, also that sentence was super long. I changed it to the following:

> Before closing, Teleport Connect will remember the tabs that you had open at the end of the previous
> session. Next time you open it, Connect will ask you if you want to reopen those tabs. If you agree,
> Connect will restore connections to all resources that were active before you closed the app.

That "If you agree" part sounds weird to me though, now that I think about it.

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 7]
```diff
@@ -78,9 +77,9 @@ log out of the cluster first.
...
 ## Restarting and Reconnecting
 
-Teleport Connect remembers the tabs that you had open and will ask you on restart if you want to
-reopen them. If you agree, Connect will restore connections to all resources that were active before
-you closed the app.
+Before closing, Teleport Connect will remember the tabs that you had open at the end of the previous
+session. Next time you open it, Connect will ask you if you want to reopen those tabs. If you agree,
+Connect will restore connections to all resources that were active before you closed the app.
 
 At the moment terminal tabs are restored only as far as the working directory is concerned. Teleport
```
Comment (avatus) [inline docs/pages/use-teleport/teleport-connect.mdx:84]:
```suggestion
At the moment, terminal tabs are restored only as far as the working directory is concerned. Teleport
```

**Review by avatus — APPROVED** [review]

**Comment by ravicious** [comment]
@ptgott Could you take a look at this PR? We're aiming to have it published by tomorrow to help with the marketing release of v10.

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 8]
```diff
@@ -0,0 +1,108 @@
+---
+title: Using Teleport Connect
+description: Using Teleport Connect
+---
+
+Teleport Connect provides easy and secure access to SSH servers and databases, with support for
+other resources such as Kubernetes clusters and application access coming in the future.
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:7]:
```suggestion
other resources such as Kubernetes clusters and applications coming in the future.
```

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+![resources tab in Teleport Connect](../../img/use-teleport/connect-cluster.png)
+
+## Installation & Upgrade
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:11]: This and other headings should be in sentence case, with only the first letter and any proper nouns capititalized

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+Head over to the [Downloads](https://goteleport.com/download/) page to download the most recent
+version of Teleport Connect. As of now, Teleport Connect supports macOS only. Double-click the
+downloaded .dmg file and drag the Teleport Connect icon to the Applications folder.
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:15]:
```suggestion
downloaded `.dmg` file and drag the Teleport Connect icon to the Applications folder.
```

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+To upgrade Teleport Connect to a newer version, download the .dmg file with that version and follow
+the installation steps.
+
+## Interface Overview
```
[the remark this answers is not in the document]
Comment (ptgott) [reply docs/pages/use-teleport/teleport-connect.mdx:19]: I think "User Interface" would make sense for this heading

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+To upgrade Teleport Connect to a newer version, drag the new version to the Applications folder.
+
+## Interface Overview
+
+![interface overview of Teleport Connect](../../img/use-teleport/connect-ui-overview.png)
+
+The top bar of Teleport Connect consists of:
+
+- The profile selector (the top right) allowing you to switch between profiles on different Teleport
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:25]:
```suggestion
- The **profile selector** (the top right), which allows you to switch between profiles on different Teleport
```

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+  clusters as well as log in or out of the clusters.
+- The connection list (the top left) showing recent connections, allowing you to seamlessly switch
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:27]:
```suggestion
- The **connection list** (the top left) showing recent connections, allowing you to seamlessly switch
```

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+  between them.
+- The command bar (in the middle) which launches any command you input in a new terminal tab.
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:29]:
```suggestion
- The **command bar** (in the middle), which launches any command you input in a new terminal tab.
```

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+- The cluster selector (to the left of the command bar) shows up only if there are leaf clusters
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:30]:
```suggestion
- The **cluster selector** (to the left of the command bar), which shows up only if you have set up Trusted Clusters and there are leaf clusters
```

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+  connected to the root cluster. It lets you browse leaf cluster resources. It also changes
+  which cluster the commands in the command bar are going to target.
+
+## Connecting to an SSH Server
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:34]: It would be great to have more UI screenshots in the sections below this point:
- Choosing an SSH server to connect to
- Choosing a database to connect to
- A relevant screenshot for the "GUI client" section
- A relevant screenshot for the "CLI Client" section
- Choosing a cluster to connect to (in "Connecting to multiple clusters")

If adding these screenshots won't be possible in the time we have, we can open a GitHub issue to do this later.

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+
+### With a CLI Client
+
+The database connection tab shows the command that can be used to connect to the database. You can
+modify the database name of the connection and then click the Run button to open a new
+terminal tab with that command executed.
+
+## Connecting to Multiple Clusters
+
+After you add your first cluster, you can keep adding more by opening the profile selector in the
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:72]: What does "After you add your first cluster" mean? Add your first leaf cluster to your root cluster? Or does this relate to the UI?

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 18]
```diff
@@ -0,0 +1,108 @@
...
+At the moment Teleport Connect supports only one user per cluster. To log in as a different user,
+log out of the cluster first.
+
+## Restarting and Reconnecting
+
+Before closing, Teleport Connect will remember the tabs that you had open at the end of the session.
+Next time you open the app, Connect will ask you if you want to reopen those tabs. If you agree,
+Connect will restore connections to all resources that were active before you closed the app.
+
+At the moment, terminal tabs are restored only as far as the working directory is concerned. Teleport
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:84]: > At the moment, terminal tabs are restored only as far as the working directory is concerned.

Is there another way to phrase "as far as ... is concerned" here?

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 19]
```diff
@@ -0,0 +1,108 @@
...
+Connect doesn't attempt to re-execute commands that were in progress when the app was closed.
+
+## Troubleshooting
+
+In an attempt to resolve issues with a cluster you can try logging out of it, closing the app and
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:89]: What kinds of issues would this be?

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 20]
```diff
@@ -0,0 +1,108 @@
...
+To submit an issue, click the Submit Feedback button at the bottom right (the speech bubble symbol)
+and follow the *Submit a Bug* link.
+
+Be sure to attach logs which can be found under `~/Library/Application Support/Teleport
+Connect/logs`. The version of the app can be found in the app menu under the About Teleport Connect
+menu item.
+
+### Updating local shell environment
+
+Teleport Connect updates the environment variables on app restart and not when starting a new shell
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:107]: > the environment variables

Which environment variables does this refer to?

File: docs/pages/use-teleport/teleport-connect.mdx (hunk 9 above)
Comment (ravicious) [reply docs/pages/use-teleport/teleport-connect.mdx:11]: > This and other headings should be in sentence case, with only the first letter and any proper nouns capititalized

Thanks! I wasn't sure and I usually saw most of my US-based peers use title case for those.

File: docs/pages/use-teleport/teleport-connect.mdx (hunk 17 above)
Comment (ravicious) [reply docs/pages/use-teleport/teleport-connect.mdx:72]: > What does "After you add your first cluster" mean? Add your first leaf cluster to your root cluster? Or does this relate to the UI?

What it really refers to is logging in to a cluster. Tbh our UI right now says "Add another cluster" but what it really should say is just "Log in to another cluster". I'll update this part along with the UI.

File: docs/pages/use-teleport/teleport-connect.mdx (hunk 19 above)
Comment (ravicious) [reply docs/pages/use-teleport/teleport-connect.mdx:89]: I was thinking of two types of issues:

1. A bug which results in the UI getting into a state which renders the app unusable, except for the top bar. For example, https://github.com/gravitational/webapps.e/issues/277.
2. An external change to the configuration of the cluster which renders the current profile unusable with the cluster.
   * It's hard for me to think of an example off the top of my head. But I've ran into issues in development where changing something about my Teleport cluster would render the profile stored by tsh & Connect invalid with no possibility of logging in again.
   * Usually that would be a result of doing something related to CA certs maybe, something that would be rather unlikely to happen in a live cluster but still possible.

I can try to do a general description of those issues here.

File: docs/pages/use-teleport/teleport-connect.mdx (hunk 20 above)
Comment (ravicious) [reply docs/pages/use-teleport/teleport-connect.mdx:107]: It refers to whatever env variables are set as a result of running your shell startup files.

This should really say just "local shell environment" rather than "environment variables" specifically.

File: docs/pages/use-teleport/teleport-connect.mdx (hunk 16 above)
Comment (ravicious) [reply docs/pages/use-teleport/teleport-connect.mdx:34]: > If adding these screenshots won't be possible in the time we have, we can open a GitHub issue to do this later.

Yeah, unfortunately I definitely won't have time to add those screenshots right now.

In general, we wanted keep the screenshots to minimum as we plan to substantially change the UI in the future. But that probably won't happen until another month or two if not more.

File: docs/pages/use-teleport/teleport-connect.mdx (hunk 17 above)
Comment (ravicious) [reply docs/pages/use-teleport/teleport-connect.mdx:72]: Actually, I'll change just the docs for now.

File: docs/pages/use-teleport/teleport-connect.mdx [hunk 21]
```diff
@@ -0,0 +1,117 @@
...
+1. Open a tab with cluster resources by clicking on the plus symbol at the end of the tab bar. You
+   can also press `Cmd + T` to achieve the same result.
+2. Select the Databases section.
+3. Look for the database server you wish to connect to and click the Connect button to the right.
+4. Select or enter the database user you with to use and press `Enter`.
+5. A new tab will open with a new connection established between your device and the database
+   server.
+
+This connection will remain active until you click the Close Connection button or close Teleport
+Connect. The port number will persist between app restarts – you can set up your favorite client
```
Comment (ptgott) [inline docs/pages/use-teleport/teleport-connect.mdx:56]:
```suggestion
Connect. The port number will persist between app restarts—you can set up your favorite client
```
I think this looks better as an em dash (minor nit)

File: docs/pages/use-teleport/teleport-connect.mdx (hunk 16 above)
Comment (ptgott) [reply docs/pages/use-teleport/teleport-connect.mdx]: Makes sense! We can hold off on the screenshots for this release

**Review by ptgott — APPROVED** [review]
